pkg/identity_codec/lib/identity_codec.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:convert';
+
+class _IdentityConverter<T> extends Converter<T, T> {
+  @override
+  T convert(T input) => input;
+}
+
+/// A [Codec] that converts from T to T doing absolutely nothing.
+///
+/// This is the identity codec, the one that passes input to output when
+/// decoding and encoding. This codec is mostly useful when implementing generic
+/// code that can be fused with a [Codec] and you need an identity codec for the
+/// base case.
+///
+/// Note, that when fused with another [Codec] the identity codec disppears.
+class IdentityCodec<T> extends Codec<T, T> {
```

Comment:
Instead of having a generic identity codec, wouldn't it be better to specify a raw type (e.g. `List<int>` or `Uint8List`) and require a codec for that? Especially if the value type is not immutable, it could cause inconsistency between a serialization-based cache impl and an in-memory identity-based one.
I have a package working with `Uint8List` called `package:buffer`, and it can be used to efficiently work with different custom buffers.

Reply:
If all of them are using `List<int>`, then starting with `List<int>` seems to be better to me. I can barely imagine why anyone would use the `IdentityCodec` otherwise, it doesn't seem to be that useful to publish it as a separate package.

Reply:
For the record, I'm hoping to land `IdentityCodec` in https://github.com/dart-lang/convert/pull/17, otherwise I'll probably publish elsewhere..